stream_utils.py

```diff
@@ -0,0 +1,48 @@
+from amaranth import Module, Signal
+from amaranth.lib.wiring import Component, In, Out
+from amaranth.lib import stream
+
+
+class LimitForwarder(Component):
+    """
+    Forwards a limited number of bytes from one stream to another, then stops.
+
+    TODO: Documentation, parameterization
+    """
+
+    inbound: In(stream.Signature(8))
+    outbound: Out(stream.Signature(8))
+
+    count: In(9)
+    start: In(1)
+    done: Out(1)
+
+    def elaborate(self, _platform):
+        m = Module()
+
+        countdown = Signal(9)
+
+        with m.FSM():
+            with m.State("idle"):
+                m.next = "idle"
+                m.d.comb += self.done.eq(1)
+                with m.If(self.start):
+                    m.next = "run"
+                    m.d.sync += countdown.eq(self.count)
+            with m.State("run"):
+                m.next = "run"
+                m.d.comb += self.done.eq(0)
+                with m.If(countdown == 0):
+                    m.next = "idle"
+                with m.Else():
+                    # Doesn't want to connect()?
```

Comment:
This looks like pretty much the example output from `connect(m, self.inbound, self.outbound)`: https://amaranth-lang.org/docs/amaranth/latest/stdlib/wiring.html#reusable-interfaces  What's the error you see?

Reply:
Looked a bit, and I didn't see anything obvious. :/